java/libphonenumber/test/com/google/i18n/phonenumbers/PhoneNumberUtilTest.java

```diff
@@ -2770,6 +2786,11 @@ public void testIsNumberMatchMatches() throws Exception {
                  phoneUtil.isNumberMatch("+64 3 331-6005 extn 1234", "+6433316005#1234"));
     assertEquals(PhoneNumberUtil.MatchType.EXACT_MATCH,
                  phoneUtil.isNumberMatch("+64 3 331-6005 ext. 1234", "+6433316005;1234"));
+    // Test numbers with extension with modem control characters in them
+    assertEquals(PhoneNumberUtil.MatchType.SHORT_NSN_MATCH,
```

Comment:
SHORT_NSN_MATCH doesn't seem quite right, but maybe it is the best here.

We could return EXACT_MATCH if we specify that such differences are ignored. But most users of EXACT_MATCH may depend on exact equivalence of the PhoneNumber object / formatted result.

It's not worth introducing a new enum value for differences in the commands in the extension.

Another alternative would be if the parse(...) method drops the command characters. I.e. the extension as stored in the resulting PhoneNumber object would never include command characters.

1. What do you think of that option? I probably haven't thought this through.

2. SHORT_NSN_MATCH seems reasonable. What do you think?